### PR TITLE
Add apic timer

### PR DESCRIPTION
**Pull request by Keyboard-Slayer**

### CODE REVIEW

File: sources/arch/x86_64/pit.h [hunk 1]
```diff
@@ -1,9 +1,12 @@
 #pragma once
 
 #define PIT_FREQUENCY 1193182
+#define PIT_MSEC 1000
```
Comment (Supercip971) [inline sources/arch/x86_64/pit.h:4]: use micro or mili maybe ?

File: sources/arch/x86_64/pit_counter.s [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+PIT_DATA_PORT0 equ 0x40
+
+pit_read_counter:
+    mov al, 0
+    out PIT_MODE_COMMAND, al 
+
+    in al, PIT_DATA_PORT0
+    mov ah, al 
+
+    in al, PIT_DATA_PORT0
```
Comment (sleepy-monax) [inline sources/arch/x86_64/pit_counter.s:13]: Tu peut utilisé asm_in*() comme ca plus besoin d'asm :)

File: sources/arch/x86_64/pit.c [hunk 3]
```diff
@@ -9,3 +9,14 @@ void pit_initialize(int freq_divisor)
     asm_out8(PIT_DATA_PORT0, divisor & 0xFF);
     asm_out8(PIT_DATA_PORT0, (divisor >> 8) & 0xFF);
 }
+
+void pit_sleep(uint16_t ms)
+{
+    uint16_t wait_val = PIT_FREQUENCY / (ms * PIT_MSEC);
+    
+    asm_out8(PIT_MODE_COMMAND, PIT_CHANNEL1 | PIT_LOWBYTE);
```
Comment (Supercip971) [inline sources/arch/x86_64/pit.c:17]: maybe use a pit_write or a pit_read instead of raw out8

**Review by sleepy-monax — CHANGES_REQUESTED** [review]

File: sources/arch/x86_64/apic_timer.c [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+void apic_timer_initialize(void)
+{
+    lapic_write(LAPIC_REG_TIMER_DIV, APIC_TIMER_DIVIDE_BY_16);
+    lapic_write(LAPIC_REG_TIMER_INITCNT, 0xFFFFFFFF);              /* Set the value to -1 */
+
+    pit_sleep(10);
+
+    lapic_write(LAPIC_REG_LVT_TIMER, LAPIC_REGISTER_LVT_INT_MASKED);
+
+    uint32_t tickIn10ms = 0xFFFFFFFF - lapic_read(LAPIC_REG_TIMER_CURRCNT);
```
Comment (Supercip971) [inline sources/arch/x86_64/apic_timer.c:16]: tick_in_10ms maybe ?

File: sources/arch/x86_64/apic_timer.h [hunk 5]
```diff
@@ -0,0 +1,18 @@
+#pragma once 
+
+#define IRQ0 32
```
Comment (Supercip971) [inline sources/arch/x86_64/apic_timer.h:3]: add a prefix

Comment (Supercip971) [reply sources/arch/x86_64/apic_timer.h:3]: or move it to the interrupt.h file

**Review by Supercip971 — CHANGES_REQUESTED** [review]

File: sources/arch/x86_64/pit.h (hunk 1 above)
Comment (Keyboard-Slayer) [reply sources/arch/x86_64/pit.h:4]: wdym 
MS = Micro Second 
I think that MSEC good enough :thinking:

File: sources/arch/x86_64/pit.c (hunk 3 above)
Comment (Keyboard-Slayer) [reply sources/arch/x86_64/pit.c:17]: I didn't write that code x)
I've just removed magic value

Comment (Keyboard-Slayer) [reply sources/arch/x86_64/pit.c:17]: `pit_initialize` use the same architecture.
I don't see the need for a new function :thinking:

File: sources/arch/x86_64/pit.h (hunk 1 above)
Comment (sleepy-monax) [reply sources/arch/x86_64/pit.h:4]: US = Micro Second

**Pull request merged**